### ISSUE TITLE
App Crashed

### ISSUE THREAD

**Issue by ibrahimarslann59** [TheWidlarzGroup/react-native-video#3383]
# Bug

## Platform
Which player are you experiencing the problem on:
* iOS
* Android

## Environment info

Library version: 5.2.1 OR 6.0.0-alpha.9

## Error Message
I also get the same error when I set a local video or a uri as a source. In the ios 17 version, the app is crashing. the error message is as follows.This problem did not occur on lower versions

`Message: Main thread blocked by synchronous property query on not-yet-loaded property (NaturalSize) for HTTP(S) asset. This could have been a problem if this asset were being read from a slow network.`
<img width="771" alt="Screenshot 2023-11-25 at 22 57 29" src="https://github.com/react-native-video/react-native-video/assets/38531647/79cbf72a-ed55-4ac4-bf89-9d8a7bd854fc">

**Comment by ilkanozen** [comment]
same error

**Comment by Haseebshah936** [comment]
same error app crashing in my case as well

**Comment by wjbeau** [comment]
I get the same error message but it is not causing an app crash.  Are you certain that the app crash is being caused by this error?

**Comment by LazaDev99** [comment]
Did you fixed this error? I use Unity GMA for my iOS game and when i request for new interstitial or banner i have the same error in XCode. I tried with coroutines, but i think the main problem is with adMob listeners. If anyone fix this, please info.

**Comment by m4tt86** [comment]
same errore app crashing

**Comment by KrzysztofMoch** [comment]
Cloud you check if issue is present on `beta.4` ?

**Comment by chenweigh** [comment]
same +1

`react-native-video: 5.2.1`

**Comment by m4tt86** [comment]
I solved the problem! The image specified in the 'poster' attribute was sometimes loading slowly, leading to an error. I resolved this by hosting the poster image on a CDN. Additionally, instead of using the 'onLoad' event, I replaced it with the 'onReadyForDisplay' event (which is executed when the poster image disappears).

**Comment by iamazeez** [comment]
Is there any workaround to fix the issue? The app gets crash when we do any action before starting the video. What I try I hide all the back forward and pause actions and show when  onReadyForDisplay trigger.
But still the video starts after 2 second and if user do any action meanwhile the apps get crash

**Comment by Haseebshah936** [comment]
@iamazeez I want with expo-av after configuring expo modules

**Comment by iamazeez** [comment]
Are you facing the same issue in expo-av?? @Haseebshah936

**Comment by Csutkas** [comment]
Same error, any chance for a fix?

**Comment by Csutkas** [comment]
For me it crashed ios (17.x) when I left the screen, so I did a small workaround and it looks like working:
I made a state for the videoUrl, initially undefined. 
I set it with the proper url when I got it from my BE call. Playing the video.
When the user wants to left the screen, I set this state to undefined.
I render only the Video component from react-native-video if the url state is not undefined otherwise I render null.
With this my react native app not crashing.

Code snippet:
`  const renderVideo = useCallback(() => {
    return (
      <Video
        source={{ uri: videoUrl }}
        onBuffer={(b) => setLoading(b.isBuffering)}
        controls
        resizeMode='contain'
        style={styles.media}
      />
    );
  }, [videoUrl, setLoading]);`

`<View>{videoUrl ? renderVideo() : null}</View>`
